src/mercury_engine_data_structures/formats/bmssd.py

```diff
@@ -13,133 +17,245 @@
 
 from mercury_engine_data_structures import game_check
 from mercury_engine_data_structures.base_resource import BaseResource
-from mercury_engine_data_structures.common_types import CVector3D, StrId, VersionAdapter, make_vector
-from mercury_engine_data_structures.construct_extensions.strings import StaticPaddedString
-from mercury_engine_data_structures.game_check import Game
+from mercury_engine_data_structures.common_types import CVector3D, StrId, VersionAdapter, make_dict, make_vector
+from mercury_engine_data_structures.crc import crc32, crc64
+
+TransformStruct = Struct("position" / CVector3D, "rotation" / CVector3D, "scale" / CVector3D)
```

Comment:
honestly, throw this in `common_types` and just call it `Transform3D` or smth. that way if we want we could optimize it in one place, and I expect it'll have uses in other formats